### ISSUE TITLE
Celestial objects emit penetrating ethereal glow.

### ISSUE THREAD

**Issue by BioTechproject27** [Qendolin/better-clouds#62]
## Describe the bug
The clouds have this effect of appearing to scatter the moon and sun light, which is very beautiful, however that occurs even when they aren't/shouldn't be visible.
When blocked by vanilla weather fog (rain & thunder):
![2024-01-10_01 35 48](https://github.com/Qendolin/better-clouds/assets/100373243/5462777c-cfd8-4ed6-a9c3-0f813d949766)

or by a giant wall of solid rock:
![2024-01-10_01 44 40](https://github.com/Qendolin/better-clouds/assets/100373243/d5554147-1f63-43cb-99c4-d2af721c398e)

Also causes issues with mod(s), such as:
[Caelum Refabricated](https://modrinth.com/mod/caelum-refabricated), which adds a seperate moon cycle, thus causing a desync, where the actual moon has no light scattering cloud halo and instead a mystical glow appears [where the moon once was].
![2024-01-10_02 09 57](https://github.com/Qendolin/better-clouds/assets/100373243/032ac2f5-18c5-4b60-b966-eef8a2eeaafa)

Also, sth which is more of a feature request ig:
When used with [Distant Horizons](https://modrinth.com/mod/distanthorizons), clouds render infront of LOD chunks, which makes looking into the distance, especially with zoom mods or a spyglass, very weird.

 
- Mod Version: 
1.3.13+1.20.1-R

**Comment by Qendolin** [comment]
The incompatibility with Distant Horizons is already known and being tracked on their GitLab as it's an issue on their end. As for Caelum, I could add an option to disable to effect all together, because right now I don't have the time to add proper mod compatibility. As for the effect being visible when it shouldn't: For rain & thunder it's an easy fix, but with blocks I don't know yet.

**Comment by BioTechproject27** [comment]
> The incompatibility with Distant Horizons is already known and being tracked on their GitLab as it's an issue on their end.

Ah, nice

> I could add an option to disable to effect all together

I guess that could be a temporary option, especially for mods or resourcepacks that change the size/number of celestial objects.

As for adding support to such resourcepacks and perhaps also fixing the visibility through walls:
Perhaps one could calculate (and save the calcs) the "size" of objects in the skybox texture, and above a certain threshhold (of size and brightness [e.g. 16x16 pixels or so, which could represent a small moon]) have a raycast once a second or so from the skybox to the player to see if said objects edges are visible and add a glow accordingly. Could maybe even do that for the seperately moving moon texture that Caelum does (or maybe not, idk how they do it).
Ik raycasts are computationally expensive, but this is just a first idea. It would maybe make sense to have a low computation option (what is now) and a high comp option.

No worries though, take your time.
These are just minor inconveniences at best, not world breaking

**Comment by BioTechproject27** [comment]
Completely seperate question:
I noticed these dark 'dots'/grid thingy happening:
![2024-01-10_03 49 36-crop](https://github.com/Qendolin/better-clouds/assets/100373243/4ed7bd08-df75-4740-a344-6f8aa91cd0cc)

It just looks a bit odd, since clouds don't do that usually.. 
May be called a bug, but it's only really visible if one looks too closely
Why is that happening though? Is it not using basically transparent png's? Just curious

**Comment by Qendolin** [comment]
That's called dithering, and due to the way that I render the clouds, all of then must have the same opacity. So to fade them in and out I don't have a different option.

**Comment by Qendolin** [comment]
I've added an option to disable the "halo": https://modrinth.com/mod/better-clouds/version/1.3.15+1.20.1-R

**Comment by Pelegrin19** [comment]
Yeah, but there's a way to update the current position of the moon and have the halo effect well placed yet?

**Comment by Qendolin** [comment]
No, Astrocraft doesn't expose a way for me to query / get the moon position.
